arch/mips/src/pic32mx/pic32mx_lowinit.c

```diff
@@ -192,6 +247,12 @@ void pic32mx_lowinit(void)
   up_earlyserialinit();
 #endif
 
+#if CONFIG_PIC32MX_JTAG_ENABLE
```

Comment:
What is the HW default? 
Why not do this as compile time option?.

Reply:
I'm a bit confused -- I thought this was a compile-time option (with the new Kconfig)?

Reply:
Please fix these compilation errors:
```
chip/pic32mx_lowinit.c: In function 'pic32mx_disable_jtag':
Error: chip/pic32mx_lowinit.c:172:21: error: unused variable 'regval' [-Werror=unused-variable]
   register uint32_t regval;
                     ^
chip/pic32mx_lowinit.c: In function 'pic32mx_enable_jtag':
Error: chip/pic32mx_lowinit.c:199:21: error: unused variable 'regval' [-Werror=unused-variable]
   register uint32_t regval;
                     ^
chip/pic32mx_lowinit.c: In function 'pic32mx_lowinit':
Error: chip/pic32mx_lowinit.c:250:5: error: "CONFIG_PIC32MX_JTAG_ENABLE" is not defined [-Werror=undef]
 #if CONFIG_PIC32MX_JTAG_ENABLE
     ^
cc1: all warnings being treated as errors
make[1]: *** [Makefile:82: pic32mx_lowinit.o] Error 1
make[1]: Target 'libarch.a' not remade because of errors.
make: *** [tools/LibTargets.mk:152: arch/mips/src/libarch.a] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 290: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
```

Reply:
> What is the HW default? Why not do this as compile time option?.

I'm also confuse, could you please explain @davids5 ? He created an Kconfig option to disable JTAG, same config exist for STM32, and others.

Reply:
The PIC32MX defaults to enabling JTAG, so by default calling `pic32mx_enable_jtag` should be a no-op.  I did it explicitly since otherwise it was more difficult to follow the logic(double negation).  The option `PIC32MX_JTAG_ENABLE` was chosen since there is already a `PIC32MZ_JTAG_ENABLE`, but if there is a better name for it that would be good to know.

I'll fix those compiler errors in a bit.

Reply:
This may be considered a security issue. It is also a minor waste of code space. It would be better to have a compile time setting to Disable in the first code after reset, that is if it can not be disabled by a fuse setting.

Reply:
@davids5 I understood your point, but I suggest just changing in the Kconfig to "default n". Notice that STM32 and other arches also defines JTAG_ENABLE and similar

Reply:
I've configured it to disable JTAG by default.  Conditionally enabling the functions in terms of code space doesn't make sense(as any level of optimization will cause uncalled methods to not emit any code), but it might make sense to conditionally compile functions in or out to suppress compiler warnings.